### PR TITLE
[FEATURE] Add shellcheck to Github pipeline

### LINKED ISSUE #1398: Check the `runTests.sh` with shellcheck in GitHub actions
As a developer, I want the `runTests.sh` to get checked/linted on CI so it follows best practices and the CI checks avoid new warnings from getting introduced in PRs.

## Acceptance criteria

- [ ] The `runTests.sh` gets checked with shellCheck on CI.
- [ ] It is not necessary that this can be run locally.
- [ ] It is not necessary that this is run in GitLab CI.
- [ ] Rules that do not make sense for our `runTest.sh` are disabled.

## Example code from another project

https://github.com/oliverklee/helper-scripts/blob/main/.github/workflows/ci.yml#L11

We'll need to adapt paths and excluded rules.

### PR DESCRIPTION
**Pull request by cumuru**
Fixes #1398

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -304,6 +304,16 @@ jobs:
...
+  shellcheck:
+    name: Check shell scripts
+    runs-on: ubuntu-24.04
+    timeout-minutes: 2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Check scripts
+        run: |
+          cat Build/Scripts/runTests.sh && Build/Scripts/runTests.sh -s shellcheck
```
Comment (oliverklee) [inline .github/workflows/ci.yml:316]: Why the `cat`?

File: Build/Scripts/runTests.sh [hunk 2]
```diff
@@ -316,10 +317,11 @@ CONTAINER_INTERACTIVE="-it --init"
 HOST_UID=$(id -u)
 HOST_PID=$(id -g)
 USERSET=""
-SUFFIX=$(echo $RANDOM)
+SUFFIX="$RANDOM"
```
Comment (oliverklee) [inline Build/Scripts/runTests.sh:320]: Why this change? If this is to fix any warning, please let's fix it a separate PR first. Ditto for line 424/426.

File: Build/Scripts/runTests.sh [hunk 3]
```diff
@@ -437,6 +439,8 @@ mkdir -p .Build/public/typo3temp/var/tests
 
 IMAGE_PHP="ghcr.io/typo3/core-testing-$(echo "php${PHP_VERSION}" | sed -e 's/\.//'):latest"
 IMAGE_ALPINE="docker.io/alpine:3.8"
+#IMAGE_SHELLCHECK="botsudo/action-shellcheck:latest"
```
Comment (oliverklee) [inline Build/Scripts/runTests.sh:442]: Commented-out code, begone!

**Review by oliverklee — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cumuru) [reply .github/workflows/ci.yml:316]: It‘s a debugging statement. The checkout at github has side-effects from other branches, unfortunately.

**Comment by oliverklee** [comment]
```
$ ./Build/Scripts/runTests.sh -b podman -s shellcheck
Error: short-name "koalaman/shellcheck:latest" did not resolve to an alias and no unqualified-search registries are defined in "/etc/containers/registries.conf"
```

**Review by oliverklee — APPROVED** [review]
Neat! Thank you!
